test/e2e/gateway_single_spoke_test.go

```diff
@@ -190,103 +204,144 @@ var _ = Describe("Gateway single target cluster", func() {
-							Namespace: Pointer(gatewayapi.Namespace(tconfig.HubNamespace())),
+					By("creating a TLSPolicy in the hub")
+
+					tlsPolicy = &mgcv1alpha1.TLSPolicy{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      testID,
+							Namespace: tconfig.HubNamespace(),
 						},
-					},
-				}
-				err := tconfig.HubClient().Create(ctx, dnsPolicy)
-				Expect(err).ToNot(HaveOccurred())
+						Spec: mgcv1alpha1.TLSPolicySpec{
+							TargetRef: gatewayapiv1alpha2.PolicyTargetReference{
+								Group:     "gateway.networking.k8s.io",
+								Kind:      "Gateway",
+								Name:      gatewayapi.ObjectName(testID),
+								Namespace: Pointer(gatewayapi.Namespace(tconfig.HubNamespace())),
+							},
+							CertificateSpec: mgcv1alpha1.CertificateSpec{
+								IssuerRef: cmmetav1.ObjectReference{
+									Name:  "glbc-ca",
+									Kind:  "ClusterIssuer",
+									Group: "cert-manager.io",
+								},
+							},
+						},
+					}
+					err := tconfig.HubClient().Create(ctx, tlsPolicy)
+					Expect(err).ToNot(HaveOccurred())
 
-				// Wait for the DNSrecord to exists: this shouldn't be necessary but I have found out that AWS dns servers
-				// cache the "no such host" response for a period of aprox 10-15 minutes, which makes the test run for a
-				// long time.
-				By("waiting for the DNSRecord to be created in the Hub")
+					By("creating a DNSPolicy in the hub")
 
-				Eventually(func(ctx SpecContext) bool {
-					dnsrecord := &mgcv1alpha1.DNSRecord{ObjectMeta: metav1.ObjectMeta{
-						Name:      fmt.Sprintf("%s-%s", gw.Name, "https"),
-						Namespace: gw.Namespace,
-					},
+					dnsPolicy = &mgcv1alpha1.DNSPolicy{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      testID,
+							Namespace: tconfig.HubNamespace(),
+						},
+						Spec: mgcv1alpha1.DNSPolicySpec{
+							TargetRef: gatewayapiv1alpha2.PolicyTargetReference{
+								Group:     "gateway.networking.k8s.io",
+								Kind:      "Gateway",
+								Name:      gatewayapi.ObjectName(testID),
+								Namespace: Pointer(gatewayapi.Namespace(tconfig.HubNamespace())),
+							},
+						},
 					}
-					if err := tconfig.HubClient().Get(ctx, client.ObjectKeyFromObject(dnsrecord), dnsrecord); err != nil {
-						GinkgoWriter.Printf("[debug] unable to get DNSRecord: '%s'\n", err)
-						return false
+					err = tconfig.HubClient().Create(ctx, dnsPolicy)
+					Expect(err).ToNot(HaveOccurred())
+				})
+
+				AfterEach(func(ctx SpecContext) {
+					err := tconfig.SpokeClient(0).Delete(ctx, dnsPolicy)
+					Expect(err).ToNot(HaveOccurred())
+					err = tconfig.SpokeClient(0).Delete(ctx, tlsPolicy)
+					Expect(err).ToNot(HaveOccurred())
+				})
+
+				It("makes available a hostname that can be resolved and reachable through HTTPS", func(ctx SpecContext) {
+
+					// Wait for the DNSrecord to exists: this shouldn't be necessary but I have found out that AWS dns servers
+					// cache the "no such host" response for a period of aprox 10-15 minutes, which makes the test run for a
```

Comment:
Is this dns negative caching, or an AWS API thing a result of using the same host each test run, or something else?